includes/api/legacy/v2/class-wc-api-server.php

```diff
@@ -651,7 +651,8 @@ private function get_paginated_url( $page ) {
 	 * @return string
 	 */
 	public function get_raw_data() {
-		// $HTTP_RAW_POST_DATA is deprecated on PHP 5.6
+		// @codingStandardsIgnoreStart
+		// $HTTP_RAW_POST_DATA is deprecated on PHP 5.6.
 		if ( function_exists( 'phpversion' ) && version_compare( phpversion(), '5.6', '>=' ) ) {
 			return file_get_contents( 'php://input' );
```

Comment:
I'm really curious why `file_get_contents( 'php://input' );` can't be used down to 5.2. Do you know the history?

Reply:
@danielbachhuber there is all in the documentation:

http://php.net/manual/en/wrappers.php.php#wrappers.php.input

![screenshot from 2017-11-10 14-29-14](https://user-images.githubusercontent.com/1264099/32667991-92a2409e-c623-11e7-8192-82e5aa098f4d.png)

So this code covers all possible things that can go wrong, like `php://input` empty because has been open before hit this code, or `$HTTP_RAW_POST_DATA` unset because some strange serve configuration.

Reply:
Cool, thanks @claudiosanches. I missed that previously.